addons/website_slides_survey/models/slide_channel.py

```diff
@@ -2,9 +2,29 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from odoo import fields, models
+from odoo.osv import expression
 
 
 class Channel(models.Model):
     _inherit = 'slide.channel'
 
     nbr_certification = fields.Integer("Number of Certifications", compute='_compute_slides_statistics', store=True)
+
+    def _remove_membership(self, partner_ids):
+        """Unlink user_input from the slide.
+
+        Unlinking the user_input from the slide allows to keep track of the current
+        pool of attempts allowed since the user (last) joined the course, as only
+        those will have a slide_id."""
+        removed_channel_partner_domain = []
+        for channel in self:
+            removed_channel_partner_domain = expression.OR([
+                removed_channel_partner_domain,
+                [('partner_id', 'in', partner_ids),
+                 ('channel_id', '=', channel.id)]
+            ])
+        if (removed_channel_partner_domain):
+            slide_partner_ids = self.env['slide.slide.partner'].sudo().search(
+                removed_channel_partner_domain)
+            slide_partner_ids.user_input_ids.slide_id = None
```

Comment:
Can't we remove the "slide_partner_id" as well? That way it becomes a regular user input no longer linked to a membership?

If we do that, we can even simplify the line below as we don't even need to filter anymore.

Reply:
I'm a bit confused about removing the `slide_partner_id` on the user input. Isn't it even more information-reducing than removing the `invite_token`? 
Didn't we want to keep the information between the attempt and a membership? The complexity added by looking one step around to get the `slide_id` looked like it was OK for the better preservation of the record's integrity and context. How will one track which are the attempts of a user? (What's even the point of keeping the invite token if you don't know who it belongs to?)

Reply:
Well when we discussed this we agreed that the important part of archiving the course membership was to keep the course *progress*, not the actual certification attempts.

So I'm suggesting to get the same behavior as when we unlinked (deleted) the membership, as the code and use cases worked fine before then.
It's honestly simpler and makes sense, and the attempts will go back to being non-linked to a membership, just like it was before, no-one ever complained about that to my knowledge.

Reply:
@awa-odoo @flch-odoo  , I made the changes by removing the relationship between user_input and slide_partner_id, let me know what you think , thanks